File: app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/OsmNoteQuestType.java
package de.westnordost.streetcomplete.data.osmnotes;

import de.westnordost.streetcomplete.R;
import de.westnordost.streetcomplete.data.QuestType;
import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
import de.westnordost.streetcomplete.quests.note_discussion.NoteDiscussionForm;

public class OsmNoteQuestType implements QuestType
{
	@Override public AbstractQuestAnswerFragment createForm() { return new NoteDiscussionForm(); }
	@Override public int getIcon() { return R.drawable.ic_quest_notes; }
	@Override public int getTitle() { return R.string.quest_noteDiscussion_title; }

	@Override public int getDefaultDisabledMessage() { return 0; }
}

File: app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/OsmNoteQuestType.kt
package de.westnordost.streetcomplete.data.osmnotes

import de.westnordost.streetcomplete.R
import de.westnordost.streetcomplete.data.QuestType
import de.westnordost.streetcomplete.quests.note_discussion.NoteAnswer
import de.westnordost.streetcomplete.quests.note_discussion.NoteDiscussionForm

class OsmNoteQuestType : QuestType<NoteAnswer> {
    override fun getTitleReplacements(tags: Map<String, String>, featureName: Lazy<String?>): Array<String?> {
        return arrayOf()
    }

    override val icon = R.drawable.ic_quest_notes
    override val title = R.string.quest_noteDiscussion_title

    override fun createForm() = NoteDiscussionForm()
}
